src/App.tsx

```diff
@@ -1,19 +1,140 @@
 import React from 'react';
 import './App.css';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
 
 import users from './api/users';
+import todos from './api/todos';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
-};
+const preparedTodos: Todo[] = todos.map(todo => {
+  const user = users.find(person => person.id === todo.userId) || null;
+
+  return { ...todo, user };
+});
+
+interface State {
+  todos: Todo[];
+  query: string;
+  users: string;
+  userSelected: boolean;
+  titleAdded: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    todos: [...preparedTodos],
+    query: '',
+    users: '',
+    userSelected: true,
+    titleAdded: true,
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement> |
+  React.ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value,
+    } as any);
+
+    if (name === 'query') {
+      this.setState({
+        titleAdded: true,
+      } as any);
+    } else {
+      this.setState({
+        userSelected: true,
+      } as any);
+    }
+  };
+
+  handleSubmit = (event: any) => {
+    event.preventDefault();
+
+    const {
+      users: u,
```

Comment:
don't make such variables - then difficult to understand what is it
```suggestion
      users,
```